src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java

```diff
@@ -17,7 +36,56 @@ public class PasswordController {
     private final PasswordEntityService passwordEntityService;
 
     @GetMapping("/generate")
-    public GuestPasswordDto generatePassword(){
+    public PasswordResponseDto generatePassword(){
         return passwordEntityService.generatePassword();
     }
+
+    @PostMapping("/generate")
+    public PasswordResponseDto generateAndStorePassword(
+            @AuthenticationPrincipal AuthUserDetailsDto userDetailsDto,
+            @RequestBody PasswordRequestDto passwordRequestDto)
+
+            throws NameNotFoundException, NoSuchPaddingException, IllegalBlockSizeException,
```

Comment:
Too many exceptions
Maybe you should catch them on service level and re-throw some custom exception

Reply:
fixed